pyro/infer/mcmc/hmc.py

```diff
@@ -323,8 +326,11 @@ def sample(self, params):
 
         return z.copy()
 
-    def diagnostics(self):
+    def logging(self):
         return OrderedDict([
             ("step size", "{:.2e}".format(self.step_size)),
             ("acc. rate", "{:.3f}".format(self._accept_cnt / self._t))
         ])
+
+    def diagnostics(self):
+        return {"divergences": self._divergences}
```

Comment:
Just to confirm - this will return a python list of ints that can be used for indexing into the samples?

Reply:
Yes. It seems to me that returning a mask makes it harder to see directly how many and which samples come from diverging transition. So I follow your suggestion to only returning indices. There might be other useful diagnostic infos so I don't aggregate chain_idx with sample_idx.